modules/redhen_connection/src/Plugin/Action/ChangeConnectionRoleBase.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Drupal\redhen_connection\Plugin\Action;
+
+use Drupal\Core\Entity\DependencyTrait;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\redhen_connection\Entity\ConnectionRole;
+use Drupal\views_bulk_operations\Action\ViewsBulkOperationsActionBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a base class for operations to change a connection's role.
+ */
+abstract class ChangeConnectionRoleBase extends ViewsBulkOperationsActionBase implements ContainerFactoryPluginInterface {
```

Comment:
I don't understand why we have 2 different classes to build out `ChangeConnectionRole` -- what's the intermediary `ChangeConnectionRoleBase` for? Why not just have all the code in a `class ChangeConnectionRole extends ViewsBulkOperationsActionBase implements ContainerFactoryPluginInterface`?

Reply:
This is another bit I took from the core User module, I thought there might be a good reason to have a base class since they were doing it, but at the same time thought it was overkill. I will merge this all into the one class.

Reply:
Great! easy enough to break out later.